### ISSUE TITLE
Tab Completion Executes Profile Functions - Workaround found

### ISSUE THREAD

**Issue by TJSoftware** [dahlbyk/posh-git#600]
### TLDR
Install via Chocolatey as a workaround

### System Details
- posh-git version/path: (Tried the following 0.7.1, 0.7.3, and 0.7.3.1 [via chocolatey])
- PowerShell version: 5.1.17134.165
- Git version: 2.16.2.windows.1
- Operating system name and version: Win 10 Pro 10.0.17134 Build 17134

### Issue Summary

On load of PowerShell I load a file that I keep a set of custom alias' etc. that is loaded from the `$profile`. I have this file on multiple computers. When I added this setup to a few computers recently I ran into an issue. When I hit `tab` for an auto complete on a command one of my custom methods in the imported file always executes. This would only happen on 2 machines out of several with identical files that are controlled by source control.

### Issue Details

In the past for all computers I have had to run the command `Import-Module posh-git` after installation (even though I am using a Windows machine). If I did not run `Import-Module` I would not see the enhanced CLI. I have not considered that the issue of this report, but it might be a clue as to the issue so I am including it (I.E. I did not install properly).

For only 2 of the computers after running `Import-Module` and pressing `tab` a function from the alias file would run multiple times. If I commented out that function, a different function would run. Commenting out all other functions would run the primary function that imports the alias'. The only way that I could get away from the tab issue is to not run the `Import-Module` and I could keep all other code in the alias file.

### Workaround
On one machine I had `posh-git v 0.7.1` installed. I upgraded to `0.7.3` using the powershell upgrade command `PowerShellGet\Update-Module posh-git` and the issue remained. I tested a second machine that did have post-git installed using the powershell installation command `PowerShellGet\Install-Module posh-git -Scope CurrentUser -Force` (note: I had to remove -AllowPrerelease from the Readme docs). Also, same results with the tab.

The workaround was to use Chocolatey `choco install poshgit` giving me version `0.7.3.1`. This resolves the tab issue as well as the requirement for me to run the `Import-Module` command.

**Comment by dahlbyk** [comment]
You might check what your `$PROFILE` contains now. The Chocolatey install should have added something like `Import-Module C:\tools\path\to\posh-git.psd1`.

I'm glad you found a work-around, but I'm rather curious what could have been going on. Can I ask what the function names are that were being run when you hit <kbd>tab</kbd>? I can't think of anything we're doing that would produce the behavior you describe.

> note: I had to remove -AllowPrerelease from the Readme docs

`-AllowPrerelease`, only available in newer versions of PowerShellGet, allows installing the posh-git v1.0.0 beta.

**Comment by TJSoftware** [comment]
@dahlbyk Ahh, yes it did add the `Import-Module` to my `$profile`. I did not notice that change (probably because I was doing late night coding). Thank you for the follow up.

> Can I ask what the function names are that were being run when you hit tab?

Yes, I have several alias' for git, dotnet, etc, sometimes I like to list them out. I first noticed the tab issue with my custom function `Get-Alias-Custom` that lists out all that I have created. I also tried with built in PowerShell functions such as `Get-Help` and `New-Item`. I was unable to find a pattern on the verb match to what was executed.
